cmake/external/spdlog.cmake

```diff
@@ -17,7 +23,8 @@ ExternalProject_Add(
 	-DSPDLOG_BUILD_BENCH=OFF
 	-DSPDLOG_BUILD_TESTS=OFF
 	INSTALL_COMMAND ${CMAKE_COMMAND} -E copy_directory ${spdlog_SOURCE_DIR}/include ${spdlog_INCLUDE_DIR}
-    BUILD_BYPRODUCTS ${spdlog_DIR}/${CMAKE_STATIC_LIBRARY_PREFIX}spdlog${CMAKE_STATIC_LIBRARY_SUFFIX}
+    BUILD_BYPRODUCTS ${spdlog_LIBRARY_NAME}
+    BUILD_ALWAYS 1
```

Comment:
@vigsterkr probably not best practice but it is the only way to switch from debug to release (and back) without cmake complaining and having to delete the build directory